snapcraft/meta/snap_yaml.py

```diff
@@ -177,7 +182,9 @@ def write(project: Project, prime_dir: Path, *, arch: str, version: str, grade:
     )
 
     yaml.add_representer(str, _repr_str, Dumper=yaml.SafeDumper)
-    yaml_data = snap_metadata.yaml(exclude_none=True, sort_keys=False, width=1000)
+    yaml_data = snap_metadata.yaml(
```

Comment:
When generating yaml from a pydantic model, `alias_generator = lambda s: s.replace("_", "-")` is ignored unless `by_alias=True` is passed.